ui/src/app/workflows/components/workflow-details/suspend-inputs.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+import { Parameter } from "../../../../models";
+import {Select} from 'argo-ui';
+
+interface SuspendInputProps {
+    parameters: Parameter[]
+    nodeId: string
+    setParameter: (key: string, value: string) => void
+}
+
+export const SuspendInputs = (props: SuspendInputProps) => {
+
+    const [parameters, setParameters] = React.useState(props.parameters)
+
```

Comment:
This looks wrong to me. You should not call `props.setParameter`. The behavior is unpredictable. If you need to propagate the change, add an `onParameterChanged(name,value)` to `props`. You may find if you do that, you don’t need the state.

Reply:
Yeah, worked without a state
Thanks

Reply:
Actually it is required for me to store the state.
Reason being `Select` component displays the value coming from `value` parameter always.

Wasn't able to make it work without this
We can discuss if there is a better way.